scipy/sparse/tests/test_base.py

```diff
@@ -4607,6 +4607,13 @@ def test_tocoo_tocsr_tocsc_gh19245(self):
         csc = dia.tocsc()
         assert csc.indices.dtype == np.int32
 
+    def test_mul_scalar(self):
+        # repro for gh-20434
+        m = dia_matrix([[1, 2], [0, 4]])
+        res = m * 3
+        assert isinstance(res, dia_matrix)
+        assert_array_equal(res.toarray(), [[3, 6], [0, 12]])
```

Comment:
Would be good to test the behaviour @dschult pointed out too
```suggestion
        res = m * 3
        assert isinstance(res, dia_matrix)
        assert_array_equal(res.toarray(), [[3, 6], [0, 12]])
        res = m.multiply(3)
        assert isinstance(res, dia_matrix)
        assert_array_equal(res.toarray(), [[3, 6], [0, 12]])
```

Reply:
Done, good idea.